### ISSUE TITLE
Tutorial 6 Net code structure issue

### ISSUE THREAD

**Issue by Radu2k** [jeshraghian/snntorch#140]
* Operating System: Colab

### Description

While implementing the LeakySigmoidSurrogate model presented in tutorial 6 I came across different outputs from my personal implementation following the Colab Notebook and snn.Leaky() with the same configurations.

After diving into snn.Leaky() implementation I realised that the Net forward function is not the same. 

- Current forward 

        spk = self.surrogate_func((mem - self.threshold), self.slope)  # call the Heaviside function
        reset = (spk - self.threshold).clone().detach()
        mem = self.beta * mem + input_ - reset

- snn.Leaky() forward 

        mem_shift = mem - self.threshold
        reset = self.surrogate_func(mem_shift, self.slope).clone().detach()
        mem = self.beta * (mem - reset * self.threshold) + input_
        spk = self.surrogate_func(mem_shift, self.slope)  # call the Heaviside function

As well I think that in the __init__ instead of `super(Leaky_Surrogate, self).__init__()` should be `super(LeakySigmoidSurrogate, self).__init__()`

**Comment by jeshraghian** [comment]
Thanks for flagging this. We'll look into it. 

@ridgerchu is slightly changing the structure of the Leaky neuron to simplify it; I'll look into this closer once the neuron update is done.
